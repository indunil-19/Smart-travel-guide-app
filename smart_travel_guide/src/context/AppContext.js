import React, {createContext, useReducer} from 'react';
import { initialState, reducer as AppReducer } from '../reducer/AppReducer';

export const AppContext =createContext()

export const AppProvider=({children})=>{
    const [state, dispatch]= useReducer(AppReducer, initialState)

    return(
        <AppContext.Provider  value={{state,dispatch}}>
                {children}
        </AppContext.Provider>
    )
}