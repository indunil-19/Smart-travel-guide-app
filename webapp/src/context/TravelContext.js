import React, {createContext, useReducer} from 'react';
import { initialState, Travelreducer } from '../reducer/TravelReducer';

const TravelContext =createContext()

const TravelProvider=({children})=>{
    const [state, dispatch]= useReducer(Travelreducer, initialState)

    return(
        <TravelContext.Provider  value={{state,dispatch}}>
                {children}
        </TravelContext.Provider>
    )
}

export  {TravelContext,  TravelProvider}
 