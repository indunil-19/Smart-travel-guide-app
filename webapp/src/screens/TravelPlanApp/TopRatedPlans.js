import { PlanCard } from "../../components/TravelPlanApp/planCard,"
import { Flex } from "@chakra-ui/layout"
import React,{ useState,useEffect } from "react"
 const TopRatedPlans=()=>{
    const [plans,setPlans]=useState([])
    useEffect(()=>{
            fetch("/user/getPublicTravelPlans").then(res=>res.json())
            .then(result=>{
                console.log(result)
                setPlans(result.myPlans)
            })
    },[])
    return(
        <>
        

        <Flex flexDirection="column" width="80%" p={3} justifyContent="center" alignItems="center" mx="auto" data-testid="tplans">  

                {plans && 
                    plans.map((plan , index)=>{
                        return(
                            <>
                                <Flex minWidth="md">
                                    <PlanCard  name={plan.name ? plan.name : `My plan ${index+1}`} days={plan.travelPlan[0].length} createdDate={plan.createdAt} travelPlan={plan.travelPlan}/>
                                 </Flex>
                            </>
                        )
                    })
                }
                
               
 
        </Flex>


        </>
    )
}
export default TopRatedPlans