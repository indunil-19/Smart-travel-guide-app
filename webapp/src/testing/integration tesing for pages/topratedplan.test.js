import React from "react"
import {render,screen, cleanup,waitFor} from "@testing-library/react";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";
import {TravelProvider} from "../../context/TravelContext"
import "@testing-library/jest-dom/extend-expect";
import { rest } from "msw";
import { setupServer } from "msw/node";
import 'whatwg-fetch'
import TopRatedPlans from "../../screens/TravelPlanApp/TopRatedPlans";


describe('top rated plans page test', () => {
    const server=setupServer(
        rest.get("/user/getPublicTravelPlans",(req,res,ctx)=>{
            return res(
                ctx.json({
                    myPlans:[
                        {
                    name:"plan1",
                    createdAt: "2021-10-15T11:07:53.534Z",
                    travelPlan:[ [[],[]],
                     []],
                    },{
                    name:"plan2",
                    createdAt: "2021-10-15T11:07:53.534Z",
                    travelPlan:[ [[],[]],
                        []],
                    }]
                })
            )
        })
    )
    beforeEach(()=>{      
    })
    beforeAll(() => server.listen());
    afterEach(() => server.resetHandlers());
    afterAll(() => server.close());
    afterEach(cleanup)

    it("shoud render without crashing", ()=>{
        const div = document.createElement("div");
        ReactDOM.render(<MemoryRouter><TravelProvider><TopRatedPlans /></TravelProvider></MemoryRouter>,div)
        
    })

    it("shoud render plan card correctly",async()=>{
        const div = document.createElement("div");
        render(<MemoryRouter><TravelProvider><TopRatedPlans /></TravelProvider></MemoryRouter>,div)

        waitFor(()=>{
            const planCard= screen.getByTestId("tplans")
            expect(planCard).toHaveTextContent("plan1")

         })
    })

    it("should match snapshot",  () => {
        const tree = renderer.create(<MemoryRouter><TravelProvider><TopRatedPlans/></TravelProvider></MemoryRouter>);
        expect(tree).toMatchSnapshot();
    });


     
})
