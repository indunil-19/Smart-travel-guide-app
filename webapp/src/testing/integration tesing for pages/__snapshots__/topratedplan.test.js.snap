// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`top rated plans page test should match snapshot 1`] = `
<div
  className="css-1mp0wmd"
  data-testid="tplans"
/>
`;
